plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/lightgrid/LightGrid.java

```diff
@@ -573,8 +574,9 @@ public void refreshData(boolean refreshColumns, boolean keepState, boolean fitVa
                         if (!fatColumns.isEmpty()) {
                             // Narrow fat columns on decWidth
                             int freeSpace = (clientWidth - normalWidth - getBorderWidth() - rowHeaderWidth - vScroll.getWidth())
-                                / fatColumns.size();
-                            int newFatWidth = (freeSpace > maxColumnDefWidth ? freeSpace : maxColumnDefWidth);
+                                    / fatColumns.size();
+                            int freeSpacePercent = (int) (((double) freeSpace / clientWidth) * 100);
+                            int newFatWidth = (freeSpacePercent > maxColumnDefWidth ? freeSpace : (int) (1.0 * maxColumnDefWidth / 100 * clientWidth));
```

Comment:
Confusing multiplication by `1.0`, consider casting `maxColumnDefWidth` to `double` or replacing `100` with `100.0`